jwst/lib/set_telescope_pointing.py

```diff
@@ -148,6 +144,77 @@ def update_wcs(model, default_pa_v3=0., siaf_path=None, **kwargs):
         Keyword arguments used by matrix calculation routines.
     """
 
+    # If the type of exposure is not FGS, then attempt to get pointing
+    # from telemetry.
+    if model.meta.exposure.type.lower().startswith('fgs'):
+        update_wcs_from_header(
+            model, default_pa_v3=default_pa_v3, siaf_path=siaf_path, **kwargs
+        )
+    else:
+        update_wcs_from_telem(
+            model, default_pa_v3=default_pa_v3, siaf_path=siaf_path, **kwargs
+        )
+
+
+def update_wcs_from_header(model, default_pa_v3=0., siaf_path=None, **kwargs):
+    """ Update WCS pointing from header information
+
+    For FGS-like observations, nearly all information is already populated
+    except for the CD matrix. This simply calculates that matrix.
+
+    Parameters
+    ----------
+    model : `~jwst.datamodels.DataModel`
+        The model to update.
+    default_pa_v3 : float
+        If pointing information cannot be retrieved,
+        use this as the V3 position angle.
+    siaf_path : str
+        Unused and present only for API compatibility.
+
+    kwargs: dict
+        Unused and present only for API compatibility.
+    """
+
+    logger.info('Updating WCS from headers.')
+
+    # Get position angle
+    try:
+        pa = model.meta.wcsinfo.pa_v3
```

Comment:
The `PA_V3` angle in `model.meta.wcsinfo.pa_v3` is computed in ` calc_wcs`. It's not in the SIAF. 
So this will not have the correct value. Also, perhaps it returns `None` instead of raising an `AttributeError` since it's in the core schema.

Reply:
For guider mode exposures, SDP will be getting values from the OSF (Observatory Status File) event messages for populating the RA_V1, DEC_V1, and PA_V3 keywords, rather than the usual procedure for science exposures where our scripts (like calc_wcs) compute those values. So we should already have a value of PA_V3 to work with in the input product.

Reply:
Mention this in the comments for those that forget.